src/applications/bmqstoragetool/m_bmqstoragetool_commandprocessorfactory.t.cpp

```diff
@@ -0,0 +1,74 @@
+// Copyright 2023 Bloomberg Finance L.P.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// bmqstoragetool
+#include <m_bmqstoragetool_commandprocessorfactory.h>
+
+// TEST DRIVER
+#include <mwctst_testhelper.h>
+
+// CONVENIENCE
+using namespace BloombergLP;
+using namespace m_bmqstoragetool;
+using namespace bsl;
+
+// ============================================================================
+//                                    TESTS
+// ----------------------------------------------------------------------------
+
+static void test1_breathingTest()
+// ------------------------------------------------------------------------
+// BREATHING TEST
+//
+// Concerns:
+//   Exercise the basic functionality of the component.
+//
+// Testing:
+//   Basic functionality
+// ------------------------------------------------------------------------
+{
+    mwctst::TestHelper::printTestName("BREATHING TEST");
+    // Empty parameters
+    CommandLineArguments        arguments;
+    bsl::shared_ptr<Parameters> params(
+        new (*s_allocator_p) ParametersReal(arguments, s_allocator_p),
+        s_allocator_p);
+
+    auto cmdProcessor = CommandProcessorFactory::createCommandProcessor(
```

Comment:
`auto` can't be compiled by C++03 compiler that we support

Reply:
It seems that tests are allowed to use `auto` for better readability and not compiled under old platforms. For example [in these tests](https://github.com/search?q=repo%3Abloomberg%2Fblazingmq+auto+language%3AC%2B%2B+path%3A%2F%5Esrc%5C%2Fgroups%5C%2Fmwc%5C%2F%2F+path%3A%2F%5Esrc%5C%2Fgroups%5C%2Fmwc%5C%2Fmwcc%5C%2F%2F&type=code) `auto` is used.